File: backend/Models/File/FileMetadata.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Services;

namespace Models.Files
{
    public enum FileMetadataStatus
    {
        Pending,
        Attached,
        Orphaned
    }

    [Table("FileMetadata")]
    public class FileMetadata: ICrudId
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public long FileMetadataId { get; set; }
        [NotMapped]
        public long Id { get => FileMetadataId; set => FileMetadataId = value; }
        public string VolumePath { get; set; }
        public string FileName { get; set; }

        public FileMetadataStatus FileMetadataStatus {get; set;} = FileMetadataStatus.Pending;
        public DateTime? CreationDate { get; set; }
    }

    public class FileMetadataDto 
    {
        public string FileMetadataId { get; set; }
        public string VolumePath { get; set; }
        public string FileName { get; set; }
        public FileMetadataStatus FileMetadataStatus {get; set;} = FileMetadataStatus.Pending;
        public DateTime?CreationDate { get; set; }
    }
}

File: frontend/src/app/utils/models/file-metadata.ts
/*
The card Export / Import process should be a Pass-by-Value process, rather than a Pass-by-Reference 
- If I export a card, hand that file to a friend, and he imports the card into his own library, that new card should be completely separate from my original card, just with the same content on it (card face elements, text, images, etc.). 
- If I delete or change the original, that doesn't affect the new card.
*/
export interface FileMetadata {
  fileMetadataId: string;
  volumePath: string;
  fileName: string;
  fileMetadataStatus: FileMetadataStatus;
  creationDate: Date | null;
}

export enum FileMetadataStatus {
  Pending = 'Pending',
  Attached = 'Attached',
  Orphaned = 'Orphaned'
}
